ast-nodes/modifiable-primary.dart

```diff
@@ -1,11 +1,36 @@
 import 'primary.dart';
 import '../lexer.dart';
+import '../iterator-utils.dart';
+import 'field-access.dart';
+import 'variable.dart';
+import 'index-access.dart';
+import '../syntax-error.dart';
+import 'expression.dart';
 
 /// An abstract writable entity – something that can appear in the LHS of the assignment.
 ///
 /// For example, a variable, a record field or an array slot.
 abstract class ModifiablePrimary implements Primary {
   factory ModifiablePrimary.parse(Iterable<Token> tokens) {
+    var iterator = tokens.iterator;
+    var result;
```

Comment:
Dart will incorrectly infer the type to be `dynamic`, prefer the explicit `ModifiablePrimary` type instead